src/processes.c

```diff
@@ -1226,6 +1244,55 @@ static int ps_count_fd(int pid) {
   return (count >= 1) ? count : 1;
 } /* int ps_count_fd (pid) */
 
+static int ps_get_pss(process_entry_t *ps) {
+  FILE *fh;
+  char buffer[1024];
+  char filename[64];
+  char *fields[4];
+  int numfields;
+  unsigned long pss = 0;
+
+  if (use_smaps_rollup) {
+    snprintf(filename, sizeof(filename), "/proc/%li/smaps_rollup", ps->id);
+  } else {
+    snprintf(filename, sizeof(filename), "/proc/%li/smaps", ps->id);
+  }
+
+  if ((fh = fopen(filename, "r")) == NULL) {
+    DEBUG("ps_get_pss: Failed to open file `%s'", filename);
+    return -1;
+  }
+
+  while (fgets(buffer, sizeof(buffer), fh) != NULL) {
+    char *endptr;
+    unsigned long tmp;
+
+    if (strncmp(buffer, "Pss:", 4) != 0) {
+      continue;
+    }
+
+    numfields = strsplit(buffer, fields, STATIC_ARRAY_SIZE(fields));
+    if (numfields < 2) {
+      continue;
+    }
+
+    errno = 0;
+    endptr = NULL;
+    tmp = strtoul(fields[1], &endptr, 10);
+    if ((errno == 0) && endptr != fields[1]) {
```

Comment:
This should chould `endptr` for 'nil' character, as according to `strtoul()` manual page:
> if *nptr is not '\0' but **endptr is '\0' on return, the entire string is valid

And apparently all implementations might not set errno:
> If there were no digits at all, strtoul() stores the original value of nptr in *endptr (and returns 0).
> ...
> The implementation **may** also set errno to EINVAL in case no conversion was performed (no digits seen, and 0 returned).